modules/apps/commerce/commerce-product-tax-category-web/src/main/java/com/liferay/commerce/product/tax/category/web/internal/display/context/CPTaxCategoryDisplayContext.java

```diff
@@ -172,14 +173,25 @@ _renderRequest, getPortletURL(), null,
 			_getCPTaxCategoryOrderByComparator(
 				getOrderByCol(), getOrderByType()));
 		_searchContainer.setOrderByType(getOrderByType());
-		_searchContainer.setResultsAndTotal(
-			() -> _cpTaxCategoryService.getCPTaxCategories(
-				themeDisplay.getCompanyId(), _searchContainer.getStart(),
-				_searchContainer.getEnd(),
-				_searchContainer.getOrderByComparator()),
-			_cpTaxCategoryService.getCPTaxCategoriesCount(
-				themeDisplay.getCompanyId()));
-		_searchContainer.setRowChecker(_getRowChecker());
+
+		if (!StringUtil.equals(getKeywords(), "")) {
+			_searchContainer.setResultsAndTotal(
```

Comment:
@andrea-ale-sbarra out of scope for this ticket, but I guess we should get rid of the search container here and replace it with a dataset